osu.Game.Tests/Visual/Gameplay/TestSceneHUDOverlay.cs

```diff
@@ -68,14 +75,58 @@ public void TestExternalHideDoesntAffectConfig()
             AddAssert("config unchanged", () => originalConfigValue == config.Get<bool>(OsuSetting.ShowInterface));
         }
 
-        private void createNew(Action<HUDOverlay> action = null)
+        [Test]
+        public void TestChangeHUDVisibilityOnHiddenKeyCounter()
+        {
+            createNew();
+
+            AddStep("set keycounter visible false", () =>
+            {
+                hudOverlay.KeyCounter.ConfigVisibility.Value = false;
+                hudOverlay.KeyCounter.Visible.Value = false;
+            });
+
+            AddStep("set showhud false", () => hudOverlay.ShowHud.Value = false);
+            AddUntilStep("hidetarget is hidden", () => !hideTarget.IsPresent);
+            AddAssert("key counters hidden", () => hudOverlay.KeyCounter.All(k => !k.IsPresent));
+
+            AddStep("set showhud true", () => hudOverlay.ShowHud.Value = true);
+            AddUntilStep("hidetarget is visible", () => hideTarget.IsPresent);
+            AddAssert("key counters still hidden", () => hudOverlay.KeyCounter.All(k => !k.IsPresent));
+        }
+
+        private void createNew(Action<TestHUDOverlay> action = null)
         {
             AddStep("create overlay", () =>
             {
-                Child = hudOverlay = new HUDOverlay(null, null, null, Array.Empty<Mod>());
+                Child = hudOverlay = new TestHUDOverlay();
 
                 action?.Invoke(hudOverlay);
             });
         }
+
+        private class TestHUDOverlay : HUDOverlay
+        {
+            public new TestKeyCounterDisplay KeyCounter => (TestKeyCounterDisplay)base.KeyCounter;
```

Comment:
unnecessary

Reply:
I'm not sure this and the other comments are actually addressed? Can avoid exposing with `ChildrenOfType`.